src/components/type-group/type-option/TypeOption.js

```diff
@@ -0,0 +1,59 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { generateRandomString } from '../../../utils';
+
+import styles from './TypeOption.css';
+
+class TypeOption extends Component {
+    constructor() {
+        super();
+
+        // We are using this to greatly reduce the possibility of having duplicated ids in the dom
+        this.idPrefix = `type-option-${generateRandomString()}#`;
```

Comment:
This can be declared outside of the constructor, making the constructor unnecessary

Reply:
I will change it to a functional component instead.

Reply:
Actually, I can't change it to a functional component. I will just remove the constructor as you said.